Scripts/Pointer/BasePointer.cs

```diff
@@ -0,0 +1,298 @@
+        {
+            Selected?.Invoke(e, this);
+        }
+
+        /// <summary>
+        /// The CheckValidity method checks the valid state of the pointer collision and determines if the pointer is hovering over a valid target.
+        /// </summary>
+        protected virtual void CheckValidity()
+        {
+            if (Valid)
+            {
+                if (cachedCollision.transform == null || CollisionData.transform != cachedCollision.transform)
+                {
+                    TryEmitExit();
+                    OnEntered(GetPayload(CursorData, CollisionData));
+                }
+                cachedCollision = CollisionData;
+                HoverDuration += Time.deltaTime;
+                OnHovering(GetPayload(CursorData, CollisionData));
+            }
+            else if (cachedCollision.transform != null)
+            {
+                TryEmitExit();
+            }
+
+            if (exclusionRule != null && CollisionData.transform != null)
+            {
+                Valid = !ExclusionRule.ShouldExclude(CollisionData.transform.gameObject, exclusionRule);
+            }
+        }
+
+        /// <summary>
+        /// The TryEmitExit method checks to see if the pointer is not currently colliding with a valid target and emits the Exit event.
+        /// </summary>
+        protected virtual void TryEmitExit()
+        {
+            if (cachedCollision.transform != null)
+            {
+                HoverDuration = 0f;
+                OnExited(GetPayload(CursorData, cachedCollision));
+                cachedCollision = new RaycastHit();
+            }
+        }
+
+        /// <summary>
+        /// The GetPayload method builds a valid Pointer Payload to use in the events.
+        /// </summary>
+        /// <param name="givenCursorData">A Transform containing information about the pointer cursor Transform.</param>
+        /// <param name="givenCollisionData">A RaycastHit containing information about the current pointer collision target.</param>
+        /// <returns>A PointerPayload of the pointer's current state.</returns>
+        protected virtual PointerData GetPayload(Transform givenCursorData, RaycastHit givenCollisionData)
+        {
+            PointerData e = new PointerData
+            {
+                transform = transform,
+                positionOverride = (givenCursorData != null ? (Vector3?)givenCursorData.position : null),
+                rotationOverride = (givenCursorData != null ? (Quaternion?)givenCursorData.localRotation : null),
+                localScaleOverride = (givenCursorData != null ? (Vector3?)givenCursorData.localScale : null),
+                origin = transform.position,
+                direction = transform.forward,
+                CollisionData = givenCollisionData,
+                isActive = Active,
+                isValid = Valid,
+                hoverDuration = HoverDuration
+            };
+            return e;

```

Comment:
Let's merge the creation and return lines here to prevent the need to have less meaningful variable names like `e`.

Reply:
👍 Probably look to do this everywhere too (and add to the code conventions docs)